persist/log.go

```diff
@@ -8,44 +8,42 @@ import (
 	"github.com/NebulousLabs/Sia/build"
 )
 
-// logFileWrapper wraps a log file to perform sanity checks on the Write and
-// Close functions. An error will be returned if Close is called multiple times
-// or if Write is called after Close has been called.
-type logFileWrapper struct {
-	closed bool // Flag is set after Close is called.
-	file   *os.File
+// closeableFile wraps an os.File to perform sanity checks on its Write and
+// Close methods. When the checks are enable, calls to Write or Close will
+// panic if they are called after the file has already been closed.
+type closeableFile struct {
+	*os.File
+	closed bool
 }
 
-// newLogFileWrapper returns a logFileWrapper that has been initialized with
-// the input file.
-func newLogFileWrapper(f *os.File) *logFileWrapper {
-	return &logFileWrapper{file: f}
-}
-
-// Close closes the log file wrapper.
-func (lfw *logFileWrapper) Close() error {
+// Close closes the file and sets the closed flag.
+func (cf *closeableFile) Close() error {
 	// Sanity check - close should not have been called yet.
-	if build.DEBUG && lfw.closed {
-		panic("cannot close the logger after it has been closed")
+	if build.DEBUG && cf.closed {
+		panic("cannot close the file; already closed")
+	}
+	// Ensure that all data has actually hit the disk.
+	if err := cf.Sync(); err != nil {
```

Comment:
Is the `Sync` needed? I would like to reserve calls to `Sync` for user-triggered stuff and must-be-consistent stuff.